rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -52,69 +120,54 @@ pub fn tags_to_lanes(tags: &Tags, locale: &Locale, config: &TagsToLanesConfig) -
         return Ok(spec);
     }
 
-    let oneway = tags.is("oneway", "yes") || tags.is("junction", "roundabout");
+    let oneway = Oneway::from(tags.is("oneway", "yes") || tags.is("junction", "roundabout"));
 
-    let (num_driving_fwd, num_driving_back) = driving_lane_directions(tags, locale, oneway);
-
-    let driving_lane = if tags.is("access", "no")
-        && (tags.is("bus", "yes") || tags.is("psv", "yes")) // West Seattle
-        || tags
-            .get("motor_vehicle:conditional")
-            .map(|x| x.starts_with("no"))
-            .unwrap_or(false)
-            && tags.is("bus", "yes")
-    // Example: 3rd Ave in downtown Seattle
-    {
-        LaneDesignated::Bus
-    } else {
-        LaneDesignated::Motor
-    };
+    let (forward_side, backward_side) = initial_forward_backward(tags, locale, oneway);
 
-    // These are ordered from the road center, going outwards. Most of the members of fwd_side will
-    // have Direction::Forward, but there can be exceptions with two-way cycletracks.
-    let mut fwd_side: Vec<Lane> = iter::repeat_with(|| Lane::forward(driving_lane))
-        .take(num_driving_fwd)
-        .collect();
-    let mut back_side: Vec<Lane> = iter::repeat_with(|| Lane::backward(driving_lane))
-        .take(num_driving_back)
-        .collect();
-    // TODO Fix upstream. https://wiki.openstreetmap.org/wiki/Key:centre_turn_lane
-    if tags.is("lanes:both_ways", "1") || tags.is("centre_turn_lane", "yes") {
-        fwd_side.insert(0, Lane::both(LaneDesignated::Motor));
-    }
+    // Temporary intermediate conversion
+    let (mut forward_side, mut backward_side) = (
+        forward_side
+            .into_iter()
+            .map(|lane| lane.build())
+            .collect::<Vec<_>>(),
+        backward_side
+            .into_iter()
+            .map(|lane| lane.build())
+            .collect::<Vec<_>>(),
+    );
 
     bus(
         tags,
         locale,
         oneway,
-        &mut fwd_side,
-        &mut back_side,
+        &mut forward_side,
+        &mut backward_side,
         &mut warnings,
     )?;
 
     bicycle(
         tags,
         locale,
         oneway,
-        &mut fwd_side,
-        &mut back_side,
+        &mut forward_side,
+        &mut backward_side,
         &mut warnings,
     )?;
 
-    if driving_lane == LaneDesignated::Motor {
-        parking(tags, locale, oneway, &mut fwd_side, &mut back_side);
-    }
+    // if driving_lane == LaneDesignated::Motor {
```

Comment:
Do we have a test case with parking lanes on a non-motorized road? I've never encountered one in real life or in OSM